docs/pages/database-access/guides/aws-cross-account.mdx

```diff
@@ -0,0 +1,169 @@
+AWS account and use AWS IAM roles to grant Teleport access to databases in
+external AWS accounts.
+
+## Teleport configuration
+
+The Teleport Database Service must be configured to assume an external AWS IAM
+role and, optionally, pass an external ID when it assumes that role.
+For information about when you should use an external ID, see:
+[purpose of an AWS external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html#external-id-purpose).
+
+AWS database discovery config, static database config, and dynamic database
+config all support the `assume_role_arn` and `external_id` settings.
+
+<Tabs>
+<TabItem label="AWS Discovery">
+Configure the Teleport Database Service to assume an external AWS IAM role when
+discovering AWS databases.
+
+```yaml
+# This example configuration will discover AWS RDS databases in us-west-1
+# within AWS account `222222222222` by assuming an external AWS IAM role.
+db_service:
+  enabled: "yes"
+  aws:
+    - types: ["rds"]
+      regions: ["us-west-1"]
+      assume_role_arn: "arn:aws:iam::222222222222:role/example-role"
+      external_id: "example-external-id"
+```
+
+The AWS IAM role used to discover a database will also be used by the Teleport
+Database Service to provide access to that database.
+</TabItem>
+<TabItem label="Static Config">
+Configure the Teleport Database Service to statically register an AWS database
+in an external account and proxy connections to it.
+```yaml
+# This example configuration will statically register an RDS PostgreSQL instance
+# in us-west-1 within AWS account `222222222222` by assuming an external AWS
+# IAM role.
+db_service:
+  enabled: "yes"
+  databases:
+  - name: "rds-postgres"
+    protocol: "postgres"
+      uri: "rds-postgres.abcdef012345.us-west-1.rds.amazonaws.com:5432"
+      aws:
+        assume_role_arn: "arn:aws:iam::222222222222:role/example-role"
+        external_id: "example-external-id"
+```
+</TabItem>
+<TabItem label="Dynamic Config">
+Create a dynamic database resource to dynamically register an AWS database
+in an external account and proxy connections to it.
+
+```yaml
+# This example configuration will dynamically register an RDS PostgreSQL instance
+# in us-west-1 within AWS account `222222222222`.
+# Teleport Database Service agents that match its labels with resource selectors
+# will proxy the database by assuming the configured external AWS IAM role.
+kind: db
+version: v3
+metadata:
+  name: "rds-postgres"
+  description: "Example dynamic database resource"
+  labels:
+    env: "dev"
+spec:
+  protocol: "postgres"
+  uri: "rds-postgres.abcdef012345.us-west-1.rds.amazonaws.com:5432"
+  aws:
+    # Note that account_id must match the AWS account ID in `assume_role_arn`.
+    # Dynamic database resources do not derive `account_id` from
+    # `assume_role_arn` automatically (unlike static configuration).
+    account_id: "222222222222"
+    assume_role_arn: "arn:aws:iam::222222222222:role/example-role"
```

Comment:
I missed this from the implementation part but It would be nice to align this behavior and be consistent with static configuration.  Do we have any obstacle that stops us for doing this alignment ? If not lets create a TODO ticket for that.

Reply:
issue was we didn't want to pull AWS sdk into `api`, which is what we need to align this.